craft.lic

```diff
@@ -0,0 +1,291 @@
+    elsif rank <= 300 # Tier 5  Basic
+      shape(7, 'wood brooch', 'brooch')
+    elsif rank <= 425 # Tier 6  Somewhat Challenging
+      shape(7, 'wood armband', 'armband')
+    elsif rank <= 600 # Tier 7  Challenging
+      shape(7, 'wood choker', 'choker')
+    elsif rank <= 700 # Tier 8 - Complicated
+      shape(7, 'articulated wood necklace', 'necklace')
+    elsif rank <= 850 # Tier 9 - Intricate
+      shape(7, 'wood crown', 'crown')
+    elsif rank <= 1175 # Tier 10 - Difficult
+      shape(7, 'wood comb', 'comb')
+    elsif rank <= 1400 # Tier 11 - Very Difficult
+      shape(7, 'wood haircomb', 'haircomb')
+    else # Tier 12  Extremely Difficult
+      echo('*** NOT YET IMPLEMENTED ***')
+    end
+  end
+
+  def train_forging
+    eligible = @settings.train_workorders & %w[Blacksmithing Weaponsmithing]
+    unless eligible.empty?
+      return unless money_for_training?(5000, 'Forging')
+      wait_for_script_to_complete('workorders', [eligible.first])
+      wait_for_script_to_complete('sell-loot')
+      walk_to(@training_room)
+      return
+    end
+
+    rank = DRSkill.getrank('Forging')
+    if rank <= 25 # Tier 1 - Extremely Easy
+      smith('a shallow metal cup')
+    elsif rank <= 50 # Tier 2 - Very Easy
+      smith('a short metal mug')
+    elsif rank <= 100 # Tier 3 - Easy
+      smith('a back scratcher')
+    elsif rank <= 175 # Tier 4 - Simple
+      smith('a metal ankle band')
+    elsif rank <= 300 # Tier 5 - Basic
+      smith('a metal lockpick ring')
+    elsif rank <= 425 # Tier 6 - Somewhat Challenging
+      smith('a metal armband')
+    elsif rank <= 550 # Tier 7 - Challenging
+      smith('some metal clippers')
+    elsif rank <= 700 # Tier 8 - Complicated
+      smith('some squat knitting needles')
+      # Journeyman book: triangular wire sieve, bent <metal> scissors, knobby sewing needles, squat knitting needles, narrow <metal> awl
+    elsif rank <= 850 # Tier 9 - Intricate
+      smith('serrated hide scraper')
+      # Master book: beveled wood shaper, serrated hide scraper, compact <metal> awl, compact <metal> awl, round pestle
+    elsif rank <= 1175 # Tier 10 - Difficult
+      smith('serrated scissors')
+      # Master book: slender <metal> awl, serrated scissors, grooved pestle, oblong wire sieve
+    elsif rank <= 1400 # Tier 11 - Very Difficult
+      smith('thin sewing needles')
+      # Master book: jagged wood shaper, thin sewing needles
+    else # Tier 12 - Extremely Difficult
+      smith('trapezoidal wire sieve')
+      # Master book: trapezoidal wire sieve
+    end
+  end
+
+  def train_alchemy
+    if @settings.train_workorders.include?('Remedies')
+      return unless money_for_training?(5000, 'Alchemy')
+      wait_for_script_to_complete('workorders', ['Remedies'])
+      wait_for_script_to_complete('sell-loot')
+      walk_to(@training_room)
+      return
+    end
+
+    beep
+    echo('*** NOT YET IMPLEMENTED ***')
+    @settings.crossing_training.delete('Alchemy')
```

Comment:
❓ This seems like something that a crafting script shouldn't be doing. In addition, it only modifies `@settings` for _this script_, so `crossing-training` would be unaffected.

And line 182 (maybe other lines too).

Reply:
I brought the workorder support in strictly because alchemy only works with workorders, I'll remove alchemy from this completely and we'll handle it on the crossing-training side